nilearn/decomposition/multi_pca.py

```diff
@@ -360,3 +372,44 @@ def inverse_transform(self, component_signals):
         # XXX: dealing properly with 2D/ list of 2D data?
         return [nifti_maps_masker.inverse_transform(signal)
                 for signal in component_signals]
+
+    def group_pca(self, subject_pcas):
+        """Compute the group PCA from individual subject PCAs"""
+        if self.incremental_group_pca:
+            return self._incremental_group_pca(subject_pcas)
+        else:
+            return self._in_memory_group_pca(subject_pcas)
+
+    def _incremental_group_pca(self, subject_pcas):
+        def _incremental_group_pca_impl(subject_pcas, n_components):
+            # delayed import because IncrementalPCA is only supported for
+            # sklearn versions > 0.16
+            from sklearn.decomposition.incremental_pca import IncrementalPCA
+
+            pca = IncrementalPCA(n_components=self.n_components)
+            for subject_pca in subject_pcas:
+                pca.partial_fit(subject_pca)
```

Comment:
I think we should work out a way on lazy loading subject pcas : On HCP we typically use 100 components for 1650 records, which approximately reduce the whole data set of a factor 10 (1200 samples per records) : `subject_pcas` will be around 150GB, which is too much for a laptop.

The idea would be to mini-batch subjects pcas so as to benefit from parallelization : we should partial fit on batch of `subject_pcas` of size around `n_jobs`. This would require to put Parallel call into `session_pca`, and transform this function into a generator.

Theoretically, we should then keep in memory `batch_size * single_subject_pca_size * 2`, which should keep under 2GB even on HCP.